exchanges/ftx/ftx.go

```diff
@@ -328,20 +336,62 @@ func (f *FTX) GetMarginMarketInfo(market string) ([]MarginMarketInfo, error) {
 	return r.Data, f.SendAuthHTTPRequest(exchange.RestSpot, http.MethodGet, fmt.Sprintf(marginMarketInfo, market), nil, &r)
 }
 
-// GetMarginBorrowHistory gets margin borrowing history
-func (f *FTX) GetMarginBorrowHistory() ([]MarginTransactionHistoryData, error) {
+// GetMarginBorrowHistory gets the margin borrow history data
+func (f *FTX) GetMarginBorrowHistory(startTime, endTime time.Time) ([]MarginTransactionHistoryData, error) {
 	r := struct {
 		Data []MarginTransactionHistoryData `json:"result"`
 	}{}
-	return r.Data, f.SendAuthHTTPRequest(exchange.RestSpot, http.MethodGet, marginBorrowHistory, nil, &r)
+
+	params := url.Values{}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if startTime.After(endTime) {
+			return nil, errStartTimeCannotBeAfterEndTime
+		}
+		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
+		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
+	}
+	endpoint := common.EncodeURLValues(marginBorrowHistory, params)
+	return r.Data, f.SendAuthHTTPRequest(exchange.RestSpot, http.MethodGet, endpoint, nil, &r)
+}
+
+// GetMarginMarketLendingHistory gets the markets margin lending rate history
+func (f *FTX) GetMarginMarketLendingHistory(coin string, startTime, endTime time.Time) ([]MarginTransactionHistoryData, error) {
+	r := struct {
+		Data []MarginTransactionHistoryData `json:"result"`
+	}{}
+	params := url.Values{}
+	if coin != "" {
+		params.Set("coin", coin)
+	}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if startTime.After(endTime) {
+			return nil, errStartTimeCannotBeAfterEndTime
+		}
+		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
+		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
+	}
+	endpoint := common.EncodeURLValues(marginLendingHistory, params)
+	return r.Data, f.SendAuthHTTPRequest(exchange.RestSpot, http.MethodGet, endpoint, params, &r)
 }
 
 // GetMarginLendingHistory gets margin lending history
-func (f *FTX) GetMarginLendingHistory() ([]MarginTransactionHistoryData, error) {
+func (f *FTX) GetMarginLendingHistory(coin string, startTime, endTime time.Time) ([]MarginTransactionHistoryData, error) {
```

Comment:
quite minor, but you use `currency.Code` for other requests, could you do it for this and `GetMarginMarketLendingHistory`?

Reply:
Added and also did it for the whole FTX package, also made it case insensitive in case a user uses lower case with NewCode